src/vs/workbench/browser/dnd.ts

```diff
@@ -130,6 +137,34 @@ export function extractEditorsDropData(e: DragEvent): Array<IDraggedResourceEdit
 	return editors;
 }
 
+export async function extractTreeDropData(dataTransfer: ITreeDataTransfer): Promise<Array<IDraggedResourceEditorInput>> {
+	const editors: IDraggedResourceEditorInput[] = [];
+	const resourcesKey = DataTransfers.RESOURCES.toLowerCase();
+	// Data Transfer: Resources
+	if (dataTransfer.has(resourcesKey)) {
+		try {
+			const rawResourcesData = await dataTransfer.get(resourcesKey)?.asString();
+			if (rawResourcesData) {
+				const rawResourceList = JSON.parse(rawResourcesData);
+				for (const resourceRaw of rawResourceList) {
+					if (resourceRaw.indexOf(':') > 0) { // mitigate https://github.com/microsoft/vscode/issues/124946
```

Comment:
It seems that this knowledge is now duplicate in the file, even the same comment appears. This indicates to me that the method could be made reusable in both functions to keep the knowledge in one location?

Reply:
Done!